compiler/rustc_errors/src/lib.rs

```diff
@@ -930,6 +955,40 @@ impl Handler {
     pub fn delay_as_bug(&self, diagnostic: Diagnostic) {
         self.inner.borrow_mut().delay_as_bug(diagnostic)
     }
+
+    pub fn update_unstable_expectation_id(
+        &self,
+        unstable_to_stable: &FxHashMap<LintExpectationId, LintExpectationId>,
+    ) {
+        let diags = std::mem::take(&mut self.inner.borrow_mut().unstable_expect_diagnostics);
+        if diags.is_empty() {
+            return;
+        }
+
+        let mut inner = self.inner.borrow_mut();
+        for mut diag in diags.into_iter() {
+            if let Some(mut unstable_id) = diag.level.get_expectation_id() {
+                let lint_index = unstable_id.get_lint_index();
+
+                // The unstable to stable map only maps the unstable it to a stable id
+                // the lint index is manually transferred here.
+                unstable_id.set_lint_index(None);
+                if let Some(mut stable_id) = unstable_to_stable.get(&unstable_id).map(|id| *id) {
```

Comment:
Same thing here, I think - `unstable_id` should always be successfully mapped here, so you could use `.expect` to enforce this.

Reply:
You're correct with all assumptions, I've added the corresponding tests. Thank you!